eng/native/configurecompiler.cmake

```diff
@@ -105,6 +105,10 @@ elseif (CLR_CMAKE_HOST_UNIX)
   # Use uppercase CMAKE_BUILD_TYPE for the string comparisons below
   string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
 
+  if (NOT CLR_CMAKE_HOST_ARCH_S390X)
+    add_linker_flag(-fuse-ld=lld)
+  endif ()
```

Comment:
@janvorli, should we adjust this block accordingly to support all platforms: https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/eng/native/configuretools.cmake#L66-L80

Reply:
Right, thank you, I've forgotten about this detection code

Reply:
It actually gives a match for the LD_GNU:
```
LLD 10.0.0 (compatible with GNU linkers)
```
I wonder if we need to differentiate it.

Reply:
Perhaps it would be better if the -fuse-ld=lld is moved to the same block to keep all linker-related configs in one place. Also, we can explicitly exclude `CMAKE_C_COMPILER_ID MATCHES "GNU"` from using lld (it would otherwise give errors in environment where we only have gcc-toolchain installed).


Aside, I think this line is unnecessary (and it's unique coreclr as everything is using shared or default configs): https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/pgosupport.cmake#L9

Reply:
At least the -flto is needed for PGO. The -fuse-ld=gold should not be needed when the lld linker is used, as it should also support LTO IIRC.